docs/Build.md

```diff
@@ -10,41 +10,67 @@ The design of the build system has the following main goals:
 
 * works identically on every platform
 * has minimal run-time dependencies:
-    - a working installation of `stack`
+    - `stack`
     - `git`
-* is completely functional right after simple `git clone`
-* one-stop-shop for building all executables required for using `hie` in IDEs.
+* is completely functional right after a simple `git clone` and after every `git pull`
+* one-stop-shop for building and naming all executables required for using `hie` in IDEs.
+* fails with meaningful error messages in known circumstances
+* is able to modify the environment such that `hie` can be run
+    - setup `hoogle` database
+    - setup `hlint` data-files
 
 See the project's `README` for detailed information about installing `hie`.
 
-### Tradeoffs
+### Targets
 
-#### `shake.yaml`
+The build script `install.hs` defines several targets using the `shake` build system. The targets are roughly:
 
-A `shake.yaml` is required for executing the `install.hs` file.
+* `hie-*`: builds and installs the `hie` binaries. Also renames the binaries to contain the correct version-number.
+* `build`: builds and installs `hie` binaries for all supported `ghc` versions.
+* `build-doc`: builds the hoogle-db required by `hie`
```

Comment:
Isn't hoogle-db optional? What is it used for? Documentation in tooltips?

Reply:
`hie` will display a warning on startup if it is not there or corrupt. It is therefor needed for a *complete* install of `hie`. @fendor do you know what it is used for?

The `install.hs` does not force you to execute the target: `build` installs all `hie-*` binaries, `build-doc` calls `hoogle generate` in some form.

Reply:
Hoogle is used for a bunch of stuff, like import of Modules, adding packages, tooltips and soon import-list imports as well.

Reply:
Does it make sense to always do `build-doc` then? Why `build-doc` is a separate step now? If too many features rely on build-doc should we say somewhere that we build not merely to have doc but to support many features of HIE? I was under impression that build-doc was optional because my editor (nvim) doesn't have a capability to display docs.